_posts/2022-06-15-solidity-0.8.15-release-announcement.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.8.15 Release Announcement'
+date: '2022-06-15'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.8.15](https://github.com/ethereum/solidity/releases/tag/v0.8.15)
+fixes two important bugs, improves inlining heuristics and adds ``.selector`` member for errors and events.
+
+## Important Bugs
+
+The first one is an optimizer bug that can lead to memory write operations in inline assembly being discarded if
+the result of such an operation is not read back from within the same assembly block.
+The bug can be triggered only when not using the old, non-IR compiler pipeline and happens only in assembly blocks
+that never access Solidity variables defined in the surrounding code.
+See [Optimizer Bug regarding Memory Side Effects of Inline Assembly](/2022/06/15/inline-assembly-memory-side-effects-bug/) for more information.
+
+The second bug may result in a ``.push()`` operation on a ``bytes`` array in storage appending
+bytes that are not properly zero-initialized if the array was originally copied from memory or calldata.
+This, again, can be triggered only when using the legacy code generation pipeline.
+See [Bug when Copying Dirty Bytes Arrays to Storage](/2022/06/15/dirty-bytes-array-to-storage-bug/) for more information.
+
+## Notable New Features
+
+### Improved Inlining Heuristics in Yul Optimizer
+
+Until now the the compiler used to be very conservative in deciding whether to inline a function or not.
+This was necessary due to the fact that inlining may easily increase stack pressure and lead to the dreaded "Stack too deep" error.
+
+Now that the mechanism for moving variables from stack to memory is available for the IR pipeline, we were able to relax the
+conditions necessary for inlining.
+The optimizer will now inline slightly bigger functions as long as nothing prevents the mechanism from being used.
+If you want to benefit from it, make sure that your inline assembly blocks are [memory safe](https://docs.soliditylang.org/en/latest/assembly.html#memory-safety)
+conditions, and use an EVM version newer than Homestead as the compilation target.
```

Comment:
I'm not sure whether we need to explicitly mention homestead here - I'd imagine that people really do that anyways.

Reply:
Fair point. Changed.